File: backend/app/schemas/metrics.py
from pydantic import BaseModel


class BMIMetrics(BaseModel):
    bmi: float
    category: str


class BMRMetrics(BaseModel):
    bmr: float


class TDEEMetrics(BaseModel):
    tdee: float


class MacroNutrients(BaseModel):
    calories: float
    protein: float
    carbohydrates: float
    fat: float


class HealthMetricsResponse(BaseModel):
    bmi: float
    bmi_category: str
    bmr: float
    tdee: float
    calories: float
    protein: float
    carbohydrates: float
    fat: float

File: frontend/src/types/metrics.ts
export interface HealthMetrics {
  bmi: number;
  bmi_category: string;
  bmr: number;
  tdee: number;
  calories: number;
  protein: number;
  carbohydrates: number;
  fat: number;
}

export interface ApiResponse<T> {
  message: string;
  data: T;
}